packages/combobox/src/Combobox.ts

```diff
@@ -0,0 +1,577 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+    CSSResultArray,
+    html,
+    PropertyValues,
+    TemplateResult,
+} from '@spectrum-web-components/base';
+import {
+    property,
+    query,
+} from '@spectrum-web-components/base/src/decorators.js';
+import {
+    ifDefined,
+    live,
+    repeat,
+} from '@spectrum-web-components/base/src/directives.js';
+import '../sp-combobox-item.js';
+import { ComboboxItem } from './ComboboxItem.js';
+import { Overlay } from '@spectrum-web-components/overlay';
+import '@spectrum-web-components/overlay/sp-overlay.js';
+import '@spectrum-web-components/icons-ui/icons/sp-icon-chevron100.js';
+import '@spectrum-web-components/popover/sp-popover.js';
+import '@spectrum-web-components/menu/sp-menu.js';
+import '@spectrum-web-components/menu/sp-menu-item.js';
+import '@spectrum-web-components/picker-button/sp-picker-button.js';
+import { Textfield } from '@spectrum-web-components/textfield';
+
+import styles from './combobox.css.js';
+import chevronStyles from '@spectrum-web-components/icon/src/spectrum-icon-chevron.css.js';
+import { Menu, MenuItem } from '@spectrum-web-components/menu';
+
+export type ComboboxOption = {
+    id: string;
+    value: string;
+};
```

Comment:
Question that came up for me as I work on the benchmark test: do we want to expand this API to _also_ to accept an array of strings that could get converted to `ComboboxOptions`? The `id` could match the `value`, or we could do some fancy RegEx or whatever to convert the `value` to something more multipurpose. 

The reason I ask is because I have an `String` array of a list of countries, and it's annoying for me to have to convert them into an object so I can fit it into this API. It'd be nice if I could just have the combobox do the work of converting it for me. I'm also thinking about that in case we end up using the select-only combobox as a basis for the lazy picker or something... What do you think, @Westbrook?

Reply:
Currently, we can do this with a map and have all the values in the lightDOM, but I wouldn't want that to be the only option.

```
        <sp-combobox>
            ${countryList.map(
                (value, index) => html`
                    <sp-menu-item id=${index} value=${value}>
                        ${option.value}
                    </sp-menu-item>
                `
            )}
        </sp-combobox>
```

Reply:
This is a great question for potential consumers of the element. I'm not opposed to it so much as I would be surprised to see someone with a functional data set that didn't already have not a unique identifier AND a user readable value.

In that case that there were, conditioning data to support this is also not too crazy:

```html
<sp-combobox
    .options=${countryList.map((value, index) => ({ value, index }))}
></sp-combobox>
```

This repeats the data processing more than would be appropriate, so you would probably cache a conditioned array in the parent context. That's not so different than us doing it as well, so maybe we would just do it by default.

The question will always go back to whether it actually makes sense to for consumers to actually reference values by their visible value vs always having a unique ID for them.